### PR TITLE
DOCSP-43525 - Decimal128

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-43525

### Staging Links

<li><a href=https://deploy-preview-240--mongodb-docs-csharp.netlify.app/upgrade/v3>upgrade/v3</a></li>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/upgrade.txt [hunk 1]
```diff
@@ -102,6 +102,15 @@ Version 3.0 Potential Breaking Change
 - TLS 1.0 and 1.1 are no longer supported. You must use TLS 1.2 or higher. To learn
   more about configuring TLS/SSL in the {+driver-short+}, see :ref:`<csharp-tls>`.
 
+- In previous versions of the driver, if a field had the C# data type ``Decimal128`` or
```
Comment (rustagir) [inline source/upgrade.txt:105]:
```suggestion
- In previous versions of the driver, if a field had a value of {+language+} data type ``Decimal128`` or
```

File: source/upgrade.txt [hunk 2]
```diff
@@ -102,6 +102,15 @@ Version 3.0 Potential Breaking Change
...
+  ``decimal``, the driver serialized the value as a BSON ``string`` by default. In
+  version 3.0, the driver defaults to serializing these values as BSON ``Decimal128``
+  values instead.
```
Comment (rustagir) [inline source/upgrade.txt:108]: S: flip these sentence around eg

> Serializes {+language+} ``Decimal128`` values as BSON ``Decimal128`` values. In prev versions...

File: source/upgrade.txt [hunk 3]
```diff
@@ -102,6 +102,15 @@ Version 3.0 Potential Breaking Change
...
+  To serialize a a ``decimal`` or ``Decimal128`` value as a string, 
+  apply the ``[BsonRepresentation(BsonType.String)]`` attribute to the field.
```
Comment (rustagir) [inline source/upgrade.txt:110]:
```suggestion
  To serialize a ``decimal`` or ``Decimal128`` value as a string,
  apply the ``[BsonRepresentation(BsonType.String)]`` attribute to the field.
```

Comment (rustagir) [inline source/upgrade.txt:110]: S: Instead of boradly linking to the serialization guide, consider linking to the bsonrepresentation attribute content

**Review by rustagir — CHANGES_REQUESTED** [review]
few small things

File: source/upgrade.txt [hunk 4]
```diff
@@ -102,6 +102,16 @@ Version 3.0 Potential Breaking Change
 - TLS 1.0 and 1.1 are no longer supported. You must use TLS 1.2 or higher. To learn
   more about configuring TLS/SSL in the {+driver-short+}, see :ref:`<csharp-tls>`.
 
+- By default, the driver serializes ``Decimal128`` and ``decimal`` values as BSON
+  ``Decimal128`` values. In previous versions of the driver,
+  the driver serialized these values as BSON ``string`` values
+  by default.
+  To serialize a ``decimal`` or ``Decimal128`` value as a string in v3.0, 
+  apply the ``[BsonRepresentation(BsonType.String)]`` attribute to the field.
```
Comment (rustagir) [inline source/upgrade.txt:110]: S: fix line wrapping

**Review by rustagir — APPROVED** [review]
lgtm

**Review by papafe — APPROVED** [review]
LGTM

**Pull request merged**